### PR TITLE
tutorial consistency .

### PR DESCRIPTION
**Pull request by larkin**
Updates a few segments of API to operate on new-style domain objects, which does expose some fun issues.

Also rewrote the todo API in a narrative fashion as test-api.html

bugs:
* [ ] Mutations in edit mode do not cause object to become dirty and changes are not persisted in new style views.
* [ ] can see conflicts where mutableObject.get does not reflect state since latest mutations.
* [ ] Package all CSS and templates with final build.js. (Not currently getting rendered properly).
* [ ] Package webworkers with final build
* [ ] move tutorials to a completely separate location, perhaps a separate repository, to isolate them from source code.
* [x] There is some odd behavior with objects and shared state.  Previously, the modelcacheservice guaranteed that all domain objects used a shared object reference for the model.  The new API destroys this guarantee and as such the modelcacheservice can get out of sync from the editing state of the application.  Needs to be a part of discussions about new mutation system and how we synchronize object states.  (Andrew opened #1094).

To use the tutorial you have to execute `gulp scripts` as it depends on the newly packaged mct file.

### CODE REVIEW

File: test-api.html [hunk 1]
```diff
@@ -6,8 +6,153 @@
   <script src="dist/main.js"></script>
 </head>
 <body>
-  <script>
+<script>
+  
+    // First, we're going to create the Todo List type, so that users can create
+    // todo lists.
+  
+    MCT.type('example.todo', new MCT.Type({
```
Comment (larkin) [inline test-api.html]: This seems a bit repetitive.  Could just be `MCT.type({key: 'blah', metadata: {..} ...})`.  Not sure why we need to construct types and then register them.

**Comment by akhenry** [comment]
I like the new narrative style. Putting it all in a single html file removes the magic, which is a good thing. Neatly demonstrates the clarity of the new API.

Yes, object synchronization is a very real problem. All `MutableObject`s could in principle use a wildcard listener to update the underlying model and keep themselves in sync automatically, that is the simplest approach. At the very least, `on` should probably update the property that it is listening on when listeners are fired so that within the callback a `.get()` is consistent with the passed in value.

Capitalization. Our API is inconsistently capitalized and we should settle on a standard. I think API objects should be TitleCase and API functions camelCase.

File: test-api.html (hunk 1 above)
Comment (akhenry) [inline test-api.html]: Guess it's to support subclassing and polymorphism with types?

**Comment by larkin** [comment]
Going to merge this into api-tutorials and continue working on merging in other sections of API.  I am moving the outstanding todos from this thread to the main API-tutorials PR, #999.

**Pull request merged**